### PR TITLE
Kafka Upstream plugin datatypes, schema corrections to params

### PR DESCRIPTION
**Pull request by tlblessing**
https://konghq.atlassian.net/browse/DOCS-1396 epic to add datatypes to plugins beyond the few template-autogenerated parameters.

Also as noted in this ticket https://konghq.atlassian.net/browse/DOCS-854, there shouldn't be any reference to install instructions since the plugin is bundled with KE. So I removed a vestige reference to that. Note that the obsolete install instructions still exist in the repo readme. Docs will not maintain plugin readmes in addition to plugin hub docs (@eskibars, @murillopaula).

image

Applying the same feedback from Murillo on the Kafka Logging plugin PR in progress https://github.com/Kong/docs.konghq.com/pull/2595, since they share many fields.

Made some corrections and clarifications to the parameters based on the schema.

Schema link:

https://github.com/Kong/kong-plugin-kafka-upstream/blob/master/kong/plugins/kafka-upstream/schema.lua

Direct review link:

https://deploy-preview-2597--kongdocs.netlify.app/hub/kong-inc/kafka-upstream/

### CODE REVIEW

File: app/_hub/kong-inc/kafka-upstream/index.md [hunk 1]
```diff
@@ -31,95 +31,119 @@ params:
...
+         Include the request headers in the message. At least one of these must be true:
+         `forward_method`, `forward_uri`, `forward_headers`, `forward_body`.
     - name: forward_body
-      required: false
+      required: semi
       default: "`true`"
+      datatype: boolean
       description: |
-         Include the request headers in the message.
+         Include the request body in the message. At least one of these must be true:
```
Comment (tlblessing) [inline app/_hub/kong-inc/kafka-upstream/index.md:86]: correct desc request body, not header

File: app/_data/extensions/kong-inc/kafka-upstream/versions.yml [hunk 2]
```diff
@@ -1 +1 @@
-- release: 1.3-x
+- release: 1.3-x (internal 0.2.0)
```
Comment (murillopaula) [inline app/_data/extensions/kong-inc/kafka-upstream/versions.yml:1]: same question from https://github.com/Kong/docs.konghq.com/pull/2595/#discussion_r578373044 applies

File: app/_hub/kong-inc/kafka-upstream/index.md [hunk 3]
```diff
@@ -1,6 +1,8 @@
 ---
 name: Kafka Upstream
 publisher: Kong Inc.
+version: 1.3-x
+# internal 0.0.2, only former version 0.0.1.
```
Comment (murillopaula) [inline app/_hub/kong-inc/kafka-upstream/index.md:5]: didn't get this part

`internal 0.0.2, only former version 0.0.1.`

same question from #2595 applies

File: app/_hub/kong-inc/kafka-upstream/index.md [hunk 4]
```diff
@@ -156,30 +182,28 @@ This plugin makes use of [lua-resty-kafka](https://github.com/doujiang24/lua-res
...
+2. There is no support for Authentication.
+3. There is no support for message compression.
+4. The message format is not customizable.
 
 ## Quickstart
 
-The following guidelines assume that both `Kong` and `Kafka` have been installed on your local machine:
-
-1. Install `kong-plugin-kafka-upstream` as mentioned on the "Installation" section of this readme.
+The following guidelines assume that both `Kong` and `Kafka` have been installed on your local machine.
```
Comment (murillopaula) [inline app/_hub/kong-inc/kafka-upstream/index.md:204]: If `Kong Enterprise` is installed on your local machine, then it already comes with `Kafka-Log` bundled with it.

File: app/_hub/kong-inc/kafka-upstream/index.md [hunk 5]
```diff
@@ -31,95 +33,119 @@ params:
       value_in_examples: <BOOTSTRAP_SERVERS>
       urlencode_in_examples: true
       default:
+      datatype: set of record elements
       description: |
-        List of bootstrap brokers in a `{host: host, port: port}` format.
+        Set of bootstrap brokers in a `{host: host, port: port}` list format.
```
Comment (murillopaula) [inline app/_hub/kong-inc/kafka-upstream/index.md:38]: same suggestion from https://github.com/Kong/docs.konghq.com/pull/2595/#discussion_r578388500 applies here

**Review by murillopaula — CHANGES_REQUESTED** [review]

File: app/_hub/kong-inc/kafka-upstream/index.md (hunk 3 above)
Comment (tlblessing) [reply app/_hub/kong-inc/kafka-upstream/index.md:5]: This is more of a note to self that there are plugin version issues, and that updates did not happen as they should have in the past. You'll note there is only an index file, no other versions. Need to discuss this with @eskibars and come up with a solution as to how we present and update plugin versions. Confluent only presents the most current version of a connector on their hub; perhaps we should do the same.

File: app/_data/extensions/kong-inc/kafka-upstream/versions.yml (hunk 2 above)
Comment (tlblessing) [reply app/_data/extensions/kong-inc/kafka-upstream/versions.yml:1]: > same question from #2595 applies

this was a POC but I'm removing this from the PR. The focus of this PR is to add the datatypes. The plugin versions are another lomgstanding matter that needs product management input on how to handle going forward.

File: app/_hub/kong-inc/kafka-upstream/index.md (hunk 5 above)
Comment (tlblessing) [reply app/_hub/kong-inc/kafka-upstream/index.md:38]: Done

File: app/_hub/kong-inc/kafka-upstream/index.md (hunk 4 above)
Comment (tlblessing) [reply app/_hub/kong-inc/kafka-upstream/index.md:204]: > If `Kong Enterprise` is installed on your local machine, then it already comes with `Kafka-Log` bundled with it.

Thanks for confirming. I know you meant kafka upstream, not log.

File: app/_hub/kong-inc/kafka-upstream/index.md [hunk 6]
```diff
@@ -156,30 +187,28 @@ This plugin makes use of [lua-resty-kafka](https://github.com/doujiang24/lua-res
...
+2. There is no support for Authentication.
+3. There is no support for message compression.
+4. The message format is not customizable.
 
 ## Quickstart
 
-The following guidelines assume that both `Kong` and `Kafka` have been installed on your local machine:
-
-1. Install `kong-plugin-kafka-upstream` as mentioned on the "Installation" section of this readme.
+The following guidelines assume that both `Kong` and `Kafka` have been installed on your local machine.
```
Comment (murillopaula) [inline app/_hub/kong-inc/kafka-upstream/index.md:209]:
```suggestion
The following guidelines assume that both `Kong` has been installed and that `Kafka Upstream` has been enabled.
```

File: app/_hub/kong-inc/kafka-upstream/index.md [hunk 7]
```diff
@@ -156,30 +187,28 @@ This plugin makes use of [lua-resty-kafka](https://github.com/doujiang24/lua-res
 
 When encoding request bodies, several things happen:
 
-* For requests with content-type header of `application/x-www-form-urlencoded`, `multipart/form-data`
-  or `application/json`, this plugin will pass the request body "raw" on the `body` attribute, but also try
-  to return a parse version of those arguments in `body_args`. If this parsing fails, an error message will be
-  returned and the message will not be sent.
-* If the `content-type` is not `text/plain`, `text/html`, `application/xml`, `text/xml` or `application/soap+xml`,
+* For requests with content-type header of `application/x-www-form-urlencoded`, `multipart/form-data`,
+  or `application/json`, this plugin will pass the raw request body on the `body` attribute, but also tries
```
Comment (lena-larionova) [inline app/_hub/kong-inc/kafka-upstream/index.md:191]:
```suggestion
  or `application/json`, this plugin passes the raw request body in the `body` attribute, and tries
```

File: app/_hub/kong-inc/kafka-upstream/index.md [hunk 8]
```diff
@@ -156,30 +187,28 @@ This plugin makes use of [lua-resty-kafka](https://github.com/doujiang24/lua-res
...
 ## Known issues and limitations
 
 Known limitations:
 
-1. There is no support for TLS
-2. There is no support for Authentication
-3. There is no support for message compression
-4. The message format is not customizable
+1. There is no support for TLS.
+2. There is no support for Authentication.
```
Comment (lena-larionova) [inline app/_hub/kong-inc/kafka-upstream/index.md:203]:
```suggestion
2. There is no support for authentication.
```

File: app/_hub/kong-inc/kafka-upstream/index.md [hunk 9]
```diff
@@ -156,30 +187,28 @@ This plugin makes use of [lua-resty-kafka](https://github.com/doujiang24/lua-res
...
+3. There is no support for message compression.
+4. The message format is not customizable.
 
 ## Quickstart
 
-The following guidelines assume that both `Kong` and `Kafka` have been installed on your local machine:
-
-1. Install `kong-plugin-kafka-upstream` as mentioned on the "Installation" section of this readme.
+The following guidelines assume that both `Kong` has been installed and that `Kafka Upstream` has been enabled.
```
Comment (lena-larionova) [inline app/_hub/kong-inc/kafka-upstream/index.md:209]:
```suggestion
The following guidelines assume that {{site.base_gateway}} is installed and the Kafka Upstream plugin is enabled.
```

File: app/_hub/kong-inc/kafka-upstream/index.md [hunk 10]
```diff
@@ -156,30 +187,28 @@ This plugin makes use of [lua-resty-kafka](https://github.com/doujiang24/lua-res
...
 
-2. Create `kong-upstream` topic in your `Kafka` cluster:
+1. Create a `kong-upstream` topic in your `Kafka` cluster:
```
Comment (lena-larionova) [inline app/_hub/kong-inc/kafka-upstream/index.md:211]:
```suggestion
1. Create a `kong-upstream` topic in your Kafka cluster:
```

**Review by lena-larionova — APPROVED** [review]
LGTM

**Review by murillopaula — APPROVED** [review]

**Pull request merged**